tests/jax/core_system_components/launcher_test.py

```diff
@@ -15,4 +15,381 @@
+
+    Args:
+        mock_parameter_server_fn
+    """
+    launcher = Launcher(multi_process=False)
+    with pytest.raises(Exception):
+        parameter_server = launcher.add(
+            mock_parameter_server_fn,
+            node_type=NodeType.corrier,
+            name="parameter_server_test",
+        )
+    parameter_server = launcher.add(
+        mock_parameter_server_fn,
+        node_type=NodeType.corrier,
+        name="parameter_server",
+    )
+
+    assert not hasattr(launcher, "_program")
+    assert not hasattr(launcher, "_replay_server")
+
+    assert launcher._nodes[-1] == launcher._node_dict["parameter_server"]
+    assert parameter_server == launcher._nodes[-1]
+
+    assert parameter_server == "test_parameter_server"
+
+
+def test_add_non_multi_process_two_add_calls(
+    mock_data_server_fn: Callable, mock_parameter_server_fn: Callable
+) -> None:
+    """Test calling twice add method for the case of one process
+
+    Args:
+        mock_data_server_fn
+        mock_parameter_server_fn
+    """
+    launcher = Launcher(multi_process=False)
+    data_server = launcher.add(
+        mock_data_server_fn,
+        node_type=NodeType.reverb,
+        name="data_server",
+    )
+    parameter_server = launcher.add(
+        mock_parameter_server_fn,
+        node_type=NodeType.corrier,
+        name="parameter_server",
+    )
+
+    assert not hasattr(launcher, "_program")
+
+    assert launcher._nodes == [data_server, parameter_server]
+    assert launcher._node_dict["data_server"] == data_server
+    assert launcher._node_dict["parameter_server"] == parameter_server
+
+    assert launcher._replay_server._port != None
+    assert type(launcher._replay_server._port) == int
+    assert isinstance(launcher._replay_server._server, pybind.Server)
+    assert isinstance(launcher._node_dict["data_server"], Client)
+    assert launcher._node_dict["data_server"]._server_address == (
+        f"localhost:{launcher._replay_server._port}"
+    )
+    assert isinstance(launcher._node_dict["data_server"]._client, pybind.Client)
+    assert launcher._node_dict["data_server"]._signature_cache == {}
+
+    assert launcher._node_dict["parameter_server"] == "test_parameter_server"
+
+
+def test_add_non_multi_process_two_add_same_name(
+    mock_parameter_server_fn: Callable, mock_parameter_server_second_fn: Callable
+) -> None:
+    """Test calling twice add method for two nodes with same name and for the case of one process and for node_type reverb
+
+    Args:
+        mock_data_server_fn
+    """
+    launcher = Launcher(multi_process=False)
+
+    parameter_server_1 = launcher.add(
+        mock_parameter_server_fn,
+        node_type=NodeType.corrier,
+        name="parameter_server",
+    )
+    with pytest.raises(Exception):
+        parameter_server_2 = launcher.add(
+            mock_parameter_server_second_fn,
+            node_type=NodeType.corrier,
+            name="parameter_server",
+        )
```

Comment:
Ah here this makes sense, you would expect an error from 2 of the same nodes. I assume there is some reason that multi-process allows for two nodes with the same name?

Probably worth looking into @mmorris44

Reply:
Thanks for flagging this @sash-a. I can't run launchpad locally and don't have the most experience with it, so @arnupretorius or @KaleabTessera are probably best suited to advise. Here's what I think though:
For multi-processing, it adds nodes to a `lp.Program`, grouping the node by the given name (i.e. you can have multiple executor nodes which will be grouped together), and `lp` handles this just fine. See line 97 of `mava/systems/jax/launcher.py`.

I don't see why we shouldn't support this for single-process systems though.

Reply:
I think either way works as long as it is consistent, so either both allow for multiple nodes with the same name or neither do, unless there's a reason I'm missing where it makes sense in one scenario, but doesn't make sense in another